shared/src/main/scala/pl/metastack/metarx.scala

```diff
@@ -12,4 +12,42 @@ package object metarx extends OptImplicits {
       opt := Some(t)
     }
   }
+
+  implicit class ReadChannelBooleanExtensions(rch: ReadChannel[Boolean]) {
+    def &&(other: ReadChannel[Boolean]): ReadChannel[Boolean] =  {
+      rch.flatMap (thisVal => other.map(otherVal => thisVal && otherVal) )
+    }
+    def &&(argument: Boolean): ReadChannel[Boolean] = {
+      rch.map(value => value && argument)
+    }
+
+    def ||(other: ReadChannel[Boolean]): ReadChannel[Boolean] =  {
+      rch.flatMap (thisVal => other.map(otherVal => thisVal || otherVal) )
+    }
+    def ||(argument: Boolean): ReadChannel[Boolean] = {
+      rch.map(value => value || argument)
+    }
+
+    private def getUnitChannel(input: ReadChannel[Boolean]) = {
+      val res = new RootChannel[Unit] {
+        def flush(f: Unit => Unit) {
+          input.flush(_ => f(()))
+        }
+      }
+
+      input.attach(_ => res := (()))
+
+      res
+    }
+
+    def isTrue: ReadChannel[Unit] = getUnitChannel(rch.filter(_ == true))
```

Comment:
My thought was to avoid code like this:

loginUser(usernameField.get, passwordField.get).attach {
            case true => {
                Routes.mainRoute().go()
            }
}

but maybe we could have both and call this something else? What should that be? TrueValues? trueChannel?